chain/starknet/src/data_source.rs

```diff
@@ -144,6 +175,26 @@ impl blockchain::DataSource<Chain> for DataSource {
     }
 }
 
+impl DataSource {
+    /// Returns event trigger if an event.key matches the handler.key and optionally
+    /// if event.fromAddr matches the source address. Note this only supports the default
+    /// starknet behavior of one key per event.
+    fn handler_for_event(&self, event: &codec::Event) -> Option<MappingEventHandler> {
+        return self
+            .mapping
+            .event_handlers
+            .iter()
+            .find(|handler| match event.keys.first() {
+                Some(key) => match &self.source.address {
+                    Some(address) => address == &event.from_addr && key == &handler.key(),
+                    None => key == &handler.key(),
```

Comment:
The current way of event selector comparison is problematic. Depending on the number of leading zero bytes (if any), a single selector value can have multiple equivalent representations using a byte array or `Vec`. The logic here treats `[1, 0]` and `[0, 1, 0]` as different whereas they're in fact of the same value.